### ISSUE TITLE
Che 7.11 Release 

### ISSUE THREAD

**Issue by nickboldt** [eclipse-che/che#16478]
### Release status

In parallel, the following releases can be done:

* che-theia
* che-machine-exec
    * che-plugin-registry (once che-theia and machine-exec are done)
* che-devfile registry
* che-parent, then che-docs, then che (including [quay images](https://github.com/eclipse/che/blob/master/.ci/cico_common.sh#L123-L131)) to release-candidate branch

Then in series:

* che-parent, then che-docs, then che (copy from release-candidate to release branch)
* che-operator, then chectl

- [x] che-theia (9481af8e21f3be4b2e6658854327d8fa2b390da3)
- [x] che-machine-exec
    - [x] che-plugin-registry _(depends on che-theia, che-machine-exec)_
- [x] che-devfile-registry
- [x] che-parent
    - [x] che-docs _(depends on parent)_
        - [x] che + quay images _(depends on docs and parent)_
- [x] che-operator
  - [x] https://github.com/operator-framework/community-operators/pull/1514
  - [x] https://github.com/operator-framework/community-operators/pull/1513
  - [x] chectl _(depends on all of the above)_
- [x] complete current milestone
  - [x] move incomplete *deferred* issues to backlog
  - [x] move incomplete *WIP* issues to next milestone
  - [x] close completed issues
  - [x] close milestone

| Owner | Process | Script | CI | Artifact(s) |
| --- | --- | --- | --- | --- |
| @azatsarynnyy / @nickboldt | [che-theia](https://github.com/eclipse/che-theia/blob/master/RELEASE.md) | [***MANUAL***](https://github.com/eclipse/che-theia/blob/master/RELEASE.md) | [centos](https://ci.centos.org/job/devtools-che-theia-che-release/) | [`eclipse/che-theia`](https://quay.io/eclipse/che-theia) |
| @nickboldt / @ericwill | [che-machine-exec](https://github.com/eclipse/che-machine-exec/blob/master/RELEASE.md) | [make-release.sh](https://github.com/eclipse/che-machine-exec/blob/master/make-release.sh) | [centos](https://ci.centos.org/job/devtools-che-machine-exec-release/) | [`eclipse/che-machine-exec`](https://quay.io/eclipse/che-machine-exec)| 
| @nickboldt / @ericwill | [che-plugin-registry](https://github.com/eclipse/che-plugin-registry/blob/master/RELEASE.md) | [make-release.sh](https://github.com/eclipse/che-plugin-registry/blob/master/make-release.sh) | [centos](https://ci.centos.org/job/devtools-che-plugin-registry-release/) | [`eclipse/che-plugin-registry`](https://quay.io/eclipse/che-plugin-registry)| 
| @nickboldt / @ericwill | [che-devfile-registry](https://github.com/eclipse/che-devfile-registry/blob/master/RELEASE.md) | [make-release.sh](https://github.com/eclipse/che-devfile-registry/blob/master/make-release.sh) | [centos](https://ci.centos.org/job/devtools-che-devfile-registry-release/) | [`eclipse/che-devfile-registry`](https://quay.io/eclipse/che-devfile-registry)| 
| @mkuznyetsov / @vparfonov | [che-parent](https://github.com/eclipse/che/blob/master/RELEASE.md) | [make-release.sh](https://github.com/eclipse/che-parent/blob/master/make-release.sh) | [centos](https://ci.centos.org/job/devtools-che-parent-che-release/) | [che-parent](https://search.maven.org/search?q=a:che-parent),<br/> [che.depmgt](https://search.maven.org/artifact/org.eclipse.che.depmgt/maven-depmgt-pom) |
| @mkuznyetsov / @vparfonov | [che-docs](https://github.com/eclipse/che/blob/master/RELEASE.md) | [make-release.sh](https://github.com/eclipse/che-docs/blob/master/make-release.sh) | [centos](https://ci.centos.org/job/devtools-che-docs-che-release/) | [che-docs](https://search.maven.org/search?q=a:che-docs)
| @mkuznyetsov / @vparfonov | [che](https://github.com/eclipse/che/blob/master/RELEASE.md) | [make-release.sh](https://github.com/eclipse/che/blob/master/make-release.sh) | [centos](https://ci.centos.org/job/devtools-che-che-release/) | [che.core](https://search.maven.org/search?q=che.core),<br/> [`eclipse/che-server`](https://quay.io/eclipse/che-server),<br/>[`eclipse/che-endpoint-watcher`](quay.io/eclipse/che-endpoint-watcher),<br/> [`eclipse/che-keycloak`](quay.io/eclipse/che-keycloak),<br/> [`eclipse/che-postgres`](quay.io/eclipse/che-postgres),<br/> [`eclipse/che-dev`](quay.io/eclipse/che-dev), [`eclipse/che-server`](quay.io/eclipse/che-server),<br/> [`eclipse/che-dashboard-dev`](quay.io/eclipse/che-dashboard-dev),<br/> [`eclipse/che-e2e`](quay.io/eclipse/che-e2e) |
| @tolusha| [che-operator](https://github.com/eclipse/che-operator/blob/master/RELEASE.md) | [make-release.sh](https://github.com/eclipse/che-operator/blob/master/make-release.sh) | ? | [`eclipse/che-operator`](https://quay.io/eclipse/che-operator)| 
| @tolusha| [chectl](https://github.com/che-incubator/chectl/blob/master/RELEASE.md) | [make-release.sh](https://github.com/che-incubator/chectl/blob/master/make-release.sh) | [travis](https://travis-ci.org/che-incubator/chectl) | [chectl releases](https://github.com/che-incubator/chectl/releases)

**Comment by nickboldt** [comment]
@mkuznyetsov please provide update on RC status of the Che serer and related containers / maven artifacts

**Comment by mkuznyetsov** [comment]
RC images have been built and ready for prerelease testing @dmytro-ndp 
https://ci.centos.org/job/devtools-che-che-release-candidate/12/

**Comment by dmytro-ndp** [comment]
Pre-release testing Jenkins jobs have been started on https://ci.centos.org and CRW CCI.
We are expecting test results in few hours.

**Comment by dmytro-ndp** [comment]
Current result against Multiuser Eclipse Che 7.11.0.RC, which includes:
* quay.io/eclipse/che-server:rc
* quay.io/eclipse/che-keycloak:rc
* quay.io/eclipse/che-theia:7.11.0
* quay.io/eclipse/che-theia-endpoint-runtime-binary:7.11.0
* quay.io/eclipse/che-devfile-registry:7.11.0
* quay.io/eclipse/che-plugin-registry:7.11.0
* quay.io/eclipse/che-plugin-artifacts-broker:v3.1.1
* [chectl:next](https://github.com/che-incubator/chectl/releases/download/20200401145222/chectl-linux-x64.tar.gz)
* che-operator:nightly
* centos/postgresql-96-centos7:9.6

1. Most of Java selenium tests passed on minishift 3.11 but [two tests **failed**](https://ci.centos.org/view/Devtools/job/devtools-che-rc-multiuser-integration-tests/29/console):
* `ProjectStateAfterRefreshTest.checkRestoreStateOfProjectAfterRefreshTest:77->checkFilesAreOpened:90 » Timeout`
* `ProjectStateAfterRenameWorkspaceTest.checkProjectAfterRenameWs:95->checkFilesAreOpened:111 » Timeout`
**_It requires further investigations._**

2. OS OAuth E2E tests **failed** on minishift 3.11 https://ci.centos.org/view/Devtools/job/devtools-che-rc-ocp-oauth-test/30/console
OS OAuth E2E tests have also failed starting from on nightly job https://github.com/eclipse/che/issues/16534
**_It requires further investigations._**
 
3. Rolling update tests **passed** on minishift 3.11 https://ci.centos.org/view/Devtools/job/devtools-che-rc-rolling-strategy-test/30/

4. Happy path tests, Vert.x, Java Maven, Python devfile and Git SSH flow tests **passed** 
* against MultiUser Eclipse Che 7.11.0 without TLS support on minikube 1.1.1 with operator installer:  [test report](https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/MultiUser-Che-release-check-e2e-tests-against-k8s/53/console);
* against SingleUser Eclipse Che 7.11.0 with TLS support on minikube 1.1.1 with helm installer: [devfiles tests report](https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/che-regular-tests/view/pre-release/job/Single-Che-pre-release-check-typescript-tests-against-K8S/8/), [Git SSH tests report](https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/che-regular-tests/view/pre-release/job/Single-Che-pre-release-check-typescript-tests-against-K8S/9/).

**Comment by dmytro-ndp** [comment]
There is possible blocker we should try to reproduce in Eclipse Che 7.11.0.RC: https://github.com/eclipse/che/issues/16512

**Comment by dmytro-ndp** [comment]
[Flacky Java selenium tests have passed at second try on minishift 3.11](https://ci.centos.org/view/Devtools/job/devtools-che-rc-multiuser-integration-tests/31/console):
> ProjectStateAfterRefreshTest.checkRestoreStateOfProjectAfterRefreshTest
> ProjectStateAfterRenameWorkspaceTest.checkProjectAfterRenameWs

**Comment by dmytro-ndp** [comment]
@nickboldt, @l0rd, @ibuziuk: it's seems we have a blocker for release of Eclipse Che 7.11.0.RC: https://github.com/eclipse/che/issues/16534
WDYT?

**Comment by dmytro-ndp** [comment]
- GitHub OAuth integration has been tested successfully
- we wasn't able to reproduce https://github.com/eclipse/che/issues/16512 on minishift and OCP 3.11, but it has been reproduced reliably by others, so is a blocker for release of Eclipse Che 7.11.0.RC.

**Comment by nickboldt** [comment]
I would rather release with a known bug and fix it ASAP after (then do a 7.11.1) than hold up the release past Tuesday. @l0rd  @ibuziuk would you agree?

**Comment by l0rd** [comment]
Reading the comments on the issues it looks like the issue is not always reproducible and we don't know the root cause yet. Is that right @vparfonov? I agree with @nickboldt to release 7.11.0 and a provide a fix once we have one.

**Comment by vparfonov** [comment]
Yes, it's not always reproduced, especially if QE can't reproduce it during testing we definitely should release 7.11.0

**Comment by dmytro-ndp** [comment]
@mkuznyetsov: could you, please, share the status of release?

**Comment by tolusha** [comment]
`che-operator` and `chectl` are released
